roaring/roaring.go

```diff
@@ -63,10 +64,46 @@ const (
 	maxContainerVal = 0xffff
 )
 
+type Containers interface {
+	// Get returns nil if the key does not exist.
+	Get(key uint64) *Container
+
+	// Put adds the container at key.
+	Put(key uint64, c *Container)
+
+	// PutContainerValues updates an existing container at key.
+	// If a container does not exist for key, a new one is allocated.
+	PutContainerValues(key uint64, containerType byte, n int, mapped bool)
+
+	// Remove takes the container at key out.
+	Remove(key uint64)
+
+	// GetOrCreate returns the container at key, creating a new empty container if necessary.
+	GetOrCreate(key uint64) *Container
+
+	// Clone does a deep copy of Containers, including cloning all containers contained.
+	Clone() Containers
+
+	// Last returns the highest key and associated container.
+	Last() (key uint64, c *Container)
+
+	// Size returns the number of containers stored.
+	Size() int
+
+	// Iterator returns a Contiterator which after a call to Next(), a call to Value() will
+	// return the first container at or after key. found will be true if a
+	// container is found at key.
+	Iterator(key uint64) (citer Contiterator, found bool)
+}
+
+type Contiterator interface {
```

Comment:
again, this is probably my fault, but we should at least capitalize the I in iterator, and probably just call it ContainerIterator